common/junit-platform-native/src/main/java/org/graalvm/junit/platform/config/jupiter/JupiterConfigProvider.java

```diff
@@ -0,0 +1,76 @@
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ *
+ * The above copyright notice and either this complete permission notice or at a
+ * minimum a reference to the UPL must be included in all copies or substantial
+ * portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.graalvm.junit.platform.config.jupiter;
+
+import org.graalvm.junit.platform.config.core.NativeImageConfiguration;
+import org.graalvm.junit.platform.config.core.PluginConfigProvider;
+import org.graalvm.junit.platform.config.util.AnnotationUtils;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+
+public class JupiterConfigProvider implements PluginConfigProvider {
+
+    @Override
+    public void onLoad(NativeImageConfiguration registry) {
+        String[] buildTimeInitializedClasses = new String[]{
+                "org.junit.jupiter.engine.config.EnumConfigurationParameterConverter",
+                "org.junit.jupiter.engine.descriptor.ClassTestDescriptor",
+                "org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor",
+                "org.junit.jupiter.engine.descriptor.TestFactoryTestDescriptor",
+                "org.junit.jupiter.engine.descriptor.JupiterTestDescriptor",
+                "org.junit.jupiter.engine.descriptor.MethodBasedTestDescriptor",
+                "org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor",
+                "org.junit.jupiter.engine.descriptor.TestTemplateTestDescriptor",
+                "org.junit.jupiter.engine.execution.ConditionEvaluator",
+                "org.junit.jupiter.engine.execution.ExecutableInvoker"
+        };
+        for (String className : buildTimeInitializedClasses) {
+            registry.initializeAtBuildTime(className);
+        }
+    }
+
+    @Override
+    public void onTestClassRegistered(Class<?> testClass, NativeImageConfiguration registry) {
+        AnnotationUtils.registerClassesFromAnnotationForReflection(testClass, registry, ArgumentsSource.class, source -> new Class[]{source.value()});
```

Comment:
For now, I've only migrated existing features to the new configuration mechanism. I suspect we may now support `EnumSource`s as well, however - I would mark that as "fixed" once we add tests for it in a separate PR.